src/visualizers/FactorFence.ts

```diff
@@ -0,0 +1,738 @@
+
+                    // set colour gradient for rectangle
+                    let gradient = this.palette.gradientBar
+                    if (primeIsHigh) {
+                        gradient = this.palette.gradientHighlight
+                    }
+                    if (factor.prime == this.mousePrime) {
+                        gradient = this.palette.gradientMouse
+                    }
+                    bottomColor = gradient.bottom
+                    topColor = gradient.top
+
+                    // determine where to put lower left corner
+                    const barStart = this.graphCorner.copy()
+
+                    // move over based on which term
+                    const moveOver = this.recSpace.copy()
+                    moveOver.mult(myIndex - this.first)
+
+                    // move up based on cumulative height so far
+                    const moveUp = this.sketch.createVector(0, -cumulHt)
+                    barStart.add(moveOver)
+                    barStart.add(moveUp)
+
+                    // figure out upper right corner
+                    const barDiag = this.sketch.createVector(
+                        this.recWidth,
+                        recHeight
+                    )
+
+                    // draw the rectangle
+                    this.grad_rect(
+                        barStart.x,
+                        barStart.y,
+                        barDiag.x,
+                        barDiag.y,
+                        topColor,
+                        bottomColor
+                    )
+
+                    // if the mouse is over the rectangle being drawn
+                    // then we make note of the prime factor
+                    // and term we are hovering over
+                    const testVec = this.sketch.createVector(
+                        this.sketch.mouseX,
+                        this.sketch.mouseY
+                    )
+                    testVec.mult(1 / this.scaleFactor).sub(barStart)
+                    testVec.y = testVec.y * mySign
+                    const barDiagAbs = barDiag.copy()
+                    barDiagAbs.y = -barDiagAbs.y * mySign
+                    if (
+                        testVec.x >= 0
+                        && testVec.x <= barDiagAbs.x
+                        && testVec.y <= 0
+                        && testVec.y >= barDiagAbs.y
+                    ) {
+                        this.mousePrime = factor.prime
+                        this.mouseIndex = myIndex
+                        this.mouseOn = true
+                    }
+
+                    // move upward in preparation for next bar
+                    cumulHt += recHeight
+                }
+            }
+        }
+    }
+
+    resetLoop() {
+        this.frame = 0
+        this.sketch.loop()
+    }
+
+    keyPressed() {
+        this.resetLoop()
+    }
+
+    mouseMoved() {
+        this.resetLoop()
+    }
+
+    mouseWheel(event: MouseEvent) {
+        let scaleFac = 1
```

Comment:
Oops I think I should have just made the argument to mouseWheel be a WheelEvent both in P5Visualizer.ts and here, then the instanceOf test would not be necessary. When you are making changes, could you fix that? Thanks.

Reply:
Done.

Reply:
committed